libreyolo/cli/commands/label.py

```
@@ -0,0 +1,118 @@
+        except Exception as exc:  # noqa: BLE001 - dataset/config problems
+            exit_with_error(out, "io_error", f"Failed to open dataset: {exc}")
+
+    if httpd is None:
+        exit_with_error(
+            out,
+            "io_error",
+            f"No free port found in range {port}-{port + 19}.",
+            suggestion="Pass a different port=.",
+        )
+
+    shown_url = "http://127.0.0.1:%d" % candidate if share else url
```

Ruff (0.15.15)

[warning] Use format specifiers instead of percent format

(UP031)